File: typescript/src/RollPoint.ts
export class RollPoint {
    
    value: number

    public constructor (value: number) {
        this.value = value
    }

    static of (value: number): RollPoint {
        return new RollPoint(value)
    }

}

export const SPARE = new RollPoint(-1);
export const STRIKE = new RollPoint(10);

File: java/src/main/java/it/leroymerlin/RollPoint.java
package it.leroymerlin;

public record RollPoint(Integer point) {

    public static RollPoint of(int v) {
        return new RollPoint(v);
    }

    public static final RollPoint SPARE = new RollPoint(-1);
    public static final RollPoint STRIKE = new RollPoint(10);

}
